### PR TITLE
Statistics of the HIBF layout with --verbose

### PR DESCRIPTION
**Pull request by Felix-Droop**
- Added the flag `--verbose` that leads to the command line output of additional HIBF statististics. 
- These are generated in the `hibf_model` class. The model is generated during the backtracking of the `hierarchical_binning`.
  - Side remark: Maybe we can move some other parts of the `hierarchical_binning::backtracking` function to the `hibf_model` class to make the code more comprehensible?
- I added a single CLI test for the `--verbose` flag and adjusted the `hierarchical_binning` and `simple_binning` tests.
- While doing this I encountered the bug that the false positive correction is only computed for the number of user bins and not for t_max. I fixed this and also adjusted the tests.

### CODE REVIEW

File: include/chopper/helper.hpp [hunk 1]
```diff
@@ -10,3 +14,45 @@
     return ((value + 63) >> 6) << 6;
 }
 
+//!\brief Round bytes to the appropriate unit and convert to string with unit
+[[nodiscard]] std::string byte_size_to_formatted_str(size_t bytes)
```
Comment (eseiler) [inline include/chopper/helper.hpp:18]:
```suggestion
[[nodiscard]] inline std::string byte_size_to_formatted_str(size_t bytes)
```

File: include/chopper/pack/hibf_model.hpp [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+    //!\brief The top level IBF of this HIBF model, often starting point for recursions.
+    ibf top_level_ibf;
+    //!\brief The gathered summary of statistics for each level of this HIBF.
+    robin_hood::unordered_map<size_t, level_summary> summary;
+
+public:
+    hibf_model() = delete; //!< Deleted. Would be malformed.
+    hibf_model(hibf_model const & b) = delete; //!< Defaulted.
+    hibf_model & operator=(hibf_model const &) = delete; //!< Defaulted.
```
Comment (smehringer) [inline include/chopper/pack/hibf_model.hpp:96]:
```suggestion
    hibf_model & operator=(hibf_model const &) = delete; //!< Deleted.
```

File: include/chopper/pack/hierarchical_binning.hpp [hunk 3]
```diff
@@ -27,6 +28,9 @@ class hierarchical_binning
     //!\brief The total query cost of all k-mers.
     double total_query_cost{};
 
+    //!\brief The model of the current level IBF for statistics generation
+    hibf_model::ibf * const ibf_model{nullptr};
```
Comment (smehringer) [inline include/chopper/pack/hierarchical_binning.hpp:32]: If this data object is purely to generate some statistics for us, I would not call it `model` as it suggests that it does more. It should be clear that this has nothing to do with the algorithm.

We should rename it to something like `ibf_statistics`, `ibf_stats` or `ibf_debug_info`?

File: include/chopper/pack/hierarchical_binning.hpp [hunk 4]
```diff
@@ -38,22 +42,28 @@ class hierarchical_binning
...
      * \param[in] query_cost The current cost to query all k-mers. Default: 0.0
      *
      * Each entry in the names_ and input vector respectively is considered a user bin (both vectors must have the
      * same length).
      */
-    hierarchical_binning(pack_data & data_, pack_config const & config_, double const query_cost = 0.0) :
+    hierarchical_binning(pack_data & data_, 
+                         pack_config const & config_, 
+                         hibf_model::ibf & ibf_model_,
+                         double const query_cost = 0.0) :
```
Comment (smehringer) [inline include/chopper/pack/hierarchical_binning.hpp:54]: Same here. It looks like the hierarchical binning algorithm depends on a model which is confusing.

Can the object live in `pack_data` instead?

File: src/chopper_pack.cpp [hunk 5]
```diff
@@ -79,6 +80,9 @@ int set_up_and_parse_subparser_split(seqan3::argument_parser & parser, pack_conf
                     "If given together with --determine-num-bins, all binnings up to the chosen t_max are computed "
                     "instead of stopping when the expected query costs become worse.");
 
+    parser.add_flag(config.verbose, 'v', "verbose", "Print additional statistics and information to the "
+                    "command line.");
+
```
Comment (smehringer) [inline src/chopper_pack.cpp:85]: Do you think that the information are valuable to the user who wants to pack their data?
Or is it rather for us to benchmark the algorithm?

If the former, you can leave the option but if its the latter, I would move this option to the end in the section `special options` (I added sections in some PR that is already merged) and call it `--output-statistics`

**Review by smehringer — CHANGES_REQUESTED** [review]
First round of some small but major things.

You mentioned that maybe some of the back-tracking could go into the model. I would like to keep the algorithm itself separate from any debug information output.

**Pull request closed without merging**